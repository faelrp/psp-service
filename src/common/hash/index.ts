import crypto from 'crypto';

const getTransactionHash = transaction => {
  const data = JSON.stringify(transaction);

  return crypto
    .createHash('sha256')
    .update(data)
    .digest('hex');
};

export default {
  getTransactionHash,
};
