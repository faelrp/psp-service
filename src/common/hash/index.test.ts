import crypto from 'crypto';
import hash from './index';

describe('Hash Helper', () => {
  test('should not be valid if input not respect the field type', () => {
    const createHashSpy = jest.spyOn(crypto, 'createHash');

    expect(hash.getTransactionHash({ transaction: 'test' })).toMatchSnapshot();

    expect(createHashSpy).toHaveBeenCalledTimes(1);
    expect(createHashSpy).toHaveBeenCalledWith('sha256');
  });
});
