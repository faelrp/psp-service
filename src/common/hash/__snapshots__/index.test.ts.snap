// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hash Helper should not be valid if input not respect the field type 1`] = `"8e5ec653ca4c7362376c1ba62cc36d91e0f65e6547751c03f669f67633bfdc1c"`;
